platforms/software/reporting/src/main/java/org/gradle/api/reporting/plugins/BuildDashboardPlugin.java

```diff
@@ -47,7 +47,7 @@ public void execute(final GenerateBuildDashboard buildDashboardTask) {
                 buildDashboardTask.setGroup("reporting");
 
                 DirectoryReport htmlReport = buildDashboardTask.getReports().getHtml();
-                htmlReport.getOutputLocation().convention(project.getLayout().getProjectDirectory().dir(project.provider(() -> project.getExtensions().getByType(ReportingExtension.class).file("buildDashboard").getAbsolutePath())));
+                htmlReport.getOutputLocation().convention(project.getExtensions().getByType(ReportingExtension.class).getBaseDirectory().dir("buildDashboard"));
```

Comment:
💅 Are we missing a constant for the report dir here?

Reply:
I'm not sure what you mean? buildDashboard is where the reports for this task go, so nothing else should need it.  

The task happens to have the same name, so an alternative would be to make this:
```
htmlReport.getOutputLocation().convention(project.getExtensions().getByType(ReportingExtension.class).getBaseDirectory().dir(buildDashboardTask.getName()));
```
But this would only make sense if we were configuring every GenerateBuildDashboard to use this convention.

Reply:
IMO, a test that asserts something like...
```
file("build/reports/buildDashboard/index.html").assertExists()
```
is much better than
```
file("build/reports/${BuildDashboardPlugin.DIR_NAME}/index.html").assertExists()
```
And a test that was doing something like:
```
outputLocation == file("build/reports/buildDashboard")
```
is better than
```
outputLocation == file("build/reports/${BuildDashboardPlugin.DIR_NAME}")
```

There's an added maintenance cost of changing this path, but that represents the impact that users would face if we changed this frequently.